proxy.go

```diff
@@ -421,29 +428,12 @@ func (p *Proxy) handleTunnel(session *Session) error {
 
 // handleConnect processes HTTP CONNECT requests
 func (p *Proxy) handleConnect(session *Session) error {
-	log.Debug("id=%s: connecting to host: %s", session.ID(), session.req.URL.Host)
-
-	remoteConn, err := p.connect(session, "tcp", session.RemoteAddr())
```

Comment:
This is actually quite important to make a connection attempt **before** returning response to the HTTP CONNECT request. This is how we signal that the remote host is available or not.

Reply:
however, calling the "canMITM" function first does not seem to have much impact?

Reply:
In some scenarios, the original destination address has changed after mitm, and it seems meaningless to judge whether the service is available only from the result of connect.

For example, in my usage scenario, all requests after mitm are forwarded to a proxy service for processing.

Reply:
I have replaced the logic with the previous one, you can review it.